pyspark_counter.py

```diff
@@ -0,0 +1,32 @@
+from pyspark import SparkContext
+from pyspark.streaming import StreamingContext
+from pyspark.streaming.kafka import KafkaUtils
+import json
+import argparse
+
+
+def main(topic_name, slide_interval=30, micro_secs_batch=10, output_file=None, checkpoint_file='/tmp/spark_twitter'):
+    sc = SparkContext(appName="PythonSparkStreamingKafka_RM_01")
+    sc.setLogLevel("WARN")
+
+    ssc = StreamingContext(sc, micro_secs_batch)
+    ssc.checkpoint(checkpoint_file)
+    kafkaStream = KafkaUtils.createStream(ssc, 'localhost:2181', 'spark-streaming', {topic_name:1})
+
+    parsed = kafkaStream.map(lambda v: json.loads(v[1]))
+    if output_file:
+        (parsed.map(lambda x: json.dumps(x))).saveAsTextFiles(output_file)
```

Comment:
This would generate very small files. Modify it in order to "buffer" messages and save in bigger files (like 16 MB, etc)

Reply:
Also, try to generalize it in order to change the output type (with the same JSON format, but, if we want to flush out the data to a kafka stream, or use firehose, or something else, that it wouldn't be so complicated to do it)